feat: hide internal/system topics from SHOW TOPICS (#4322)
confluentinc/ksql: docs-md/developer-guide/ksqldb-reference/show-topics.md

`localhost:9092`). SHOW TOPICS EXTENDED also displays consumer groups
and their active consumer counts.

SHOW TOPICS does not display topics considered internal, such as:

> * KSQL internal topics (i.e. the KSQL command topic)

Reviewer: ```suggestion
* KSQL internal topics, like the KSQL command topic
```

Author: Done

* Topics found in the `system.internal.topics` configuration

> SHOW ALL TOPICS will list all topics including those considered as

Reviewer: ```suggestion
SHOW ALL TOPICS lists all topics, including those considered to be
```

Author: Done

internal or found in the `ksql.internal.hidden.topics` configuration.

> 

Reviewer 2: I find the 'internal.hidden' part of the config confusing.  To me, the `internal` part is referring to KSQL's internal topics, (e.g. command topic, changelog + repartition topics, etc). But these are not controlled by the `ksql.internal.hidden.topics`!  

`ksql.internal.hidden.topics` is there to allow users to configure additional hidden topics. Hence I think `ksql.hidden.topics` would be a better name and the text above would then be:

```
`SHOW TOPICS` does not display hidden topics by default, such as:
* KSQL internal topics, like the KSQL command topic or changelog & repartition topics.
* Topics that match any pattern in the `ksql.hidden.topics` configuration.

`SHOW ALL TOPICS` lists all topics, including hidden topics.
```

IMHO this is a lot clearer.

Example
-------

```sql

> ksql> SHOW TOPICS

Reviewer: No trailing semicolon?

Author: Done

 Kafka Topic                            | Partitions | Partition Replicas
-------------------------------------------------------------------------
 default_ksql-processing-log            | 1          | 1
 pageviews                              | 1          | 1
 users                                  | 1          | 1
-------------------------------------------------------------------------
```

```sql
ksql> SHOW ALL TOPICS;

 Kafka Topic                            | Partitions | Partition Replicas
-------------------------------------------------------------------------
 _confluent-ksql-default_command_topic  | 1          | 1
 default_ksql-processing-log            | 1          | 1
 pageviews                              | 1          | 1
 users                                  | 1          | 1

> -------------------------------------------------------------------------

Reviewer 2: Maybe add an example repartition or changelog topic.

`localhost:9092`). SHOW TOPICS EXTENDED also displays consumer groups
and their active consumer counts.

`SHOW TOPICS` does not display hidden topics by default, such as:
* KSQL internal topics, like the KSQL command topic or changelog & repartition topics.

> * Topics that match any pattern in the `ksql.hidden.topics` configuration.

Author: I see both formats correctly. I used 2 bullets to mention both types of hidden topics, but your sentence is also correct. I'm not inclined to any sentence, though. 

I'll make the change if I find other changes in the code to make after the rebase.
